diff --git a/Telegram/SourceFiles/mtproto/details/mtproto_tls_socket.h b/Telegram/SourceFiles/mtproto/details/mtproto_tls_socket.h
--- a/Telegram/SourceFiles/mtproto/details/mtproto_tls_socket.h
+++ b/Telegram/SourceFiles/mtproto/details/mtproto_tls_socket.h
@@ -9,6 +9,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #include "mtproto/details/mtproto_abstract_socket.h"
 
+#include <QtNetwork/QTcpSocket>
+
 namespace MTP::details {
 
 #ifdef DESKTOP_APP_MTPROTO_PROFILE_TESTS
